web/profiles/custom/yalesites_profile/modules/custom/ys_core/src/Plugin/Block/YaleSitesBreadcrumbBlock.php

```diff
@@ -93,11 +93,13 @@ public function build() {
     ];
```

Comment:
Great. This fixes the issue. I have a suggestion for refactoring the work. Anytime I enter a logic block, I wonder about the business value. In this case the 'if' is about filtering links. This could be a separate method. I prefer breaking working into single-purpose methods. This is for reusability, readability, and hopefully for future unit testing. Take a peek at my suggestion below and we can talk about it. It seems like I am adding a lot of boillerplate but there is a reason for my madness.

```
  /**
   * {@inheritdoc}
   */
  public function build() {
    $breadcrumbs = $this->removeEmptyLinks($this->getBreadcrumbs());
    $links = [
      [
        'title' => $this->t('Home'),
        'url' => $this->urlGenerator->generateFromRoute('<front>', []),
        'is_active' => FALSE,
      ],
    ];

    foreach ($breadcrumbs as $breadcrumb) {
      array_push($links, [
        'title' => $breadcrumb->getText(),
        'url' => $breadcrumb->getUrl()->toString(),
        'is_active' => empty($breadcrumb->getUrl()->toString()),
      ]);
    }

    return [
      '#theme' => 'ys_breadcrumb_block',
      '#items' => $links,
    ];
  }

  /**
   * Get a list of breadcrumb links using the Drupal BreadcrumbBuilder.
   *
   * @return \Drupal\Core\Link[]
   *   An array of Drupal links.
   */
  protected function getBreadcrumbs(): array {
    return $this->breadcrumbManager->build($this->routeMatch)->getLinks();
  }

  /**
   * Remove empty links from a list of links.
   *
   * @param \Drupal\Core\Link[] $links
   *   An array of Drupal links.
   *
   * @return \Drupal\Core\Link[]
   *   An array of Drupal links with empty ones removed.
   */
  protected function removeEmptyLinks(array $links): array {
    return array_filter($links, function($link) {
      return $link->getText() === '';
    });
  }
```

Reply:
I honestly appreciate your feedback and teaching me new ways to do these things! Making me a better developer! I see what you're doing here. Perhaps a chat at some point about the unit testing you mentioned as I've not really done that before. I'll make those changes and then we can press forward. Thank you!